ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto

```diff
@@ -374,7 +395,7 @@ message DamlCommandDedupKey {
   string application_id = 2;
   string command_id = 3;
 }
-
+// should this be removed? how do we handle transitions?
```

Comment:
This may not be removed yet. For command deduplication related changes we'll define a process for the SDK upgrade that'll ensure that no old deduplication state interferes with the new code.

Reply:
Just double-checking this.
For the new API, to construct this message we still call the deprecated `.getDeduplicateUntil`. This means that the field is gonna be empty when `deduplication_period` is set but not in the deprecated format. 
Should this field be updated to the new deduplication period format as well? @miklos-da

Reply:
Because we're depending on the deduplication period to generate the proper min/max record times during pre-execution, we need to add the new deduplication period here as well and convert it into a maximum record time in [LedgerTimeValidator](https://github.com/digital-asset/daml/blob/main/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/LedgerTimeValidator.scala#L52).